codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerRustModule.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.shapes.UnionShape
+import software.amazon.smithy.model.traits.ErrorTrait
+import software.amazon.smithy.rust.codegen.core.rustlang.RustModule
+import software.amazon.smithy.rust.codegen.core.smithy.ModuleProvider
+import software.amazon.smithy.rust.codegen.core.smithy.traits.SyntheticInputTrait
+import software.amazon.smithy.rust.codegen.core.smithy.traits.SyntheticOutputTrait
+import software.amazon.smithy.rust.codegen.core.util.hasTrait
+
+object ServerRustModule {
+    val Error = RustModule.public("error", documentation = "All error types that operations can return. Documentation on these types is copied from the model.")
+    val Operation = RustModule.public("operation", documentation = "All operations that this crate can perform.")
+    val Model = RustModule.public("model", documentation = "Data structures used by operation inputs/outputs. Documentation on these types is copied from the model.")
+    val Input = RustModule.public("input", documentation = "Input structures for operations. Documentation on these types is copied from the model.")
+    val Output = RustModule.public("output", documentation = "Output structures for operations. Documentation on these types is copied from the model.")
+    val Types = RustModule.public("types", documentation = "Data primitives referenced by other data types.")
```

Comment:
At least on the client side, I'm intending to make this a hierarchical "database" of code generated modules. I think it's a good idea to centralize these so that you don't get conflicting documentation. I'm not sure which documentation would win in that case.